doc/.gitignore

```diff
@@ -15,6 +15,7 @@ source/data/api/doc/
 source/cluster/running-applications/job-submission/doc/
 source/serve/api/doc/
 source/workflows/api/doc/
+source/data/api/ray.data.*.rst
```

Comment:
Probably it makes sense to move the build artifacts into a separate directory, as is done for the rest of the API docs. Otherwise we end up needing to construct regexes like this to ignore them. 

It would mean just modifying the `toctree` line in `source/data/api/_autogen.rst` to put them into a separate directory. What do you think?

Reply:
these auto-generated files need to be in the same directory with the indext.rst, unfortunately, since they are added via `:include`, and `:include` only work properly on the same directory level ugg, as far as i can tell

Reply:
Ugh, there's got to be a better way than this. Can we make an upstream contribution to sphinx to allow for templates in autoclasses or something like that?

Reply:
yes let me think more about it; i'll merge this one for now